### ISSUE TITLE
Debug PiPrm test failure

### ISSUE THREAD

**Issue by marsella** [boltlabs-inc/tss-ecdsa#224]
For the test seed specified below, the `piprm::bad_secret_totient_fails()` test does not pass. Specifically, the proof generated with the bad totient verifies.

I checked the following things:
- The bad totient does not equal the correct totient
- The bad totient is not a multiple of the correct totient's factors `p` or `q`
- The challenge bytes for the good and bad totients are not the same
- The proof and verification methods seem correct to the paper

I'm not sure if this is just a lucky random number for cheaters or if there's an implementation bug or a check we're missing.

```rust
let failing_seed = [227, 82, 50, 235, 153, 149, 165, 44, 18, 223, 85, 36, 85, 79, 21, 55, 54, 47, 227, 82, 87, 106, 149, 147, 25, 184, 209, 105, 161, 109, 102, 209]
```

[Link to failing CI run](https://github.com/boltlabs-inc/tss-ecdsa/actions/runs/4386384757/jobs/7680361110#step:7:494).

## Closing criteria
- [x] Spend up to 3 hours figuring out why this test failed
- [x] Make any necessary changes

**Comment by Ra1issa** [comment]
First observation: I think this is a bug and not just an unlucky strike. I'm running the tests multiple times and they fail pretty early on in my trials.  Here are examples of other failing seeds:

`seed: [40, 99, 237, 15, 146, 76, 59, 212, 230, 252, 161, 70, 233, 153, 103, 178, 195, 92, 53, 191, 190, 36, 91, 170, 208, 110, 24, 26, 254, 130, 154, 8]`

`seed: [76, 7, 100, 241, 106, 244, 131, 173, 171, 55, 154, 135, 216, 216, 196, 98, 150, 8, 41, 77, 254, 99, 96, 147, 3, 0, 216, 157, 213, 134, 98, 0]`

`seed: [158, 142, 84, 178, 174, 139, 124, 116, 138, 237, 139, 100, 249, 122, 113, 234, 131, 178, 154, 242, 157, 72, 235, 161, 214, 56, 172, 108, 126, 213, 18, 182]`

**Comment by Ra1issa** [comment]
I don't think that `bad_secret_totient_fails` is useful/doing what we would like it to do. 

**Setup** 

The tests currently start by generating a correct Ring-Pedersen Parameters proof, which includes `s`, `t`, `lambda`, `modulus`, `totient`, such that:

- `modulus` an RSA modulus for sufficiently large primes (in this case of size 1024)
- `totient = phi(modulus)`
- `lambda <- Z_totient`
- `t := s ^ lambda` 

The test then ignores that totient and instead uniformly samples a wrong:
`badtotient <- Z_modulus`.

**Prover**


The prover then:

- Samples secret exponents `a_i <- Z_badtotient`
- Create the commitments `ta_i = t^a_i`
- The challenge bytes, a vector of 0/1 sampled uniformly at random that is as larger as the number of commitments
- And finally the responses: `z_i := a_i + lambda mod badtotient`

**Verifier**

The verifier then checks that `ta_i * s == t^z_i`.  If we expand that we get that the verifier is checking if: 
    `t^{a_i mod badtotient + lambda mod totient} == t^{a_i + lambda mod badtotient}`

The way I see it, this depends on whether `badtotient < totient` or not. Its unclear to me how the Prover can cheat by using the wrong totient function after the public parameters have been set.

**Comment by marsella** [comment]
Closed by #233.
